### ISSUE TITLE
`upcase` expression with Greek lowercase letters with accents does not remove accents

### ISSUE THREAD

**Issue by wipfli** [maplibre/maplibre-gl-js#4395]
**maplibre-gl-js version**: 4.5.0

**browser**: chrome

### Steps to Trigger Behavior

 1. Have a property with a lowercase greek string that has accents
 2. Turn it to uppercase with the `upcase` expression
 3. See how accent are not removed

### Link to Demonstration

https://jsbin.com/jukebec/edit?html,output

### Expected Behavior

```
with name = Μυρτώο Πέλαγος
['upcase', ['get', 'name']] gives ΜΥΡΤΩΟ ΠΕΛΑΓΟΣ
```

### Actual Behavior
```
with name = Μυρτώο Πέλαγος
['upcase', ['get', 'name']] gives ΜΥΡΤΏΟ ΠΈΛΑΓΟΣ
```

**Comment by wipfli** [comment]
I'm not speaking Greek so maybe this is not really an issue...

**Comment by HarelM** [comment]
The code is using javascript code to do the `toUpperCase` so I'm not sure there's anything we should do here...

**Comment by wipfli** [comment]
Yes I was thinking too that it must be using the javascript function `toUpperCase` for this. However, CSS seems to have a different behavior at least on my browser (chrome, ubuntu):

normal:
![image](https://github.com/user-attachments/assets/d5b8ddaa-3b2f-47de-b421-c3c7995a6987)

uppercase:
![image](https://github.com/user-attachments/assets/65839dd9-bf77-498e-98f2-4f63618964e5)

You can try it out here: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform

**Comment by wipfli** [comment]
JavaScript has also `toLocaleUpperCase` but I don't think that would help much here unless the user specifies the locale... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleUpperCase
